### PR TITLE
Initial SDK sample app

### PR DESCRIPTION
**Pull request by silverjam**
## Changes

Add a simple sample application/daemon for the SDK that incorporates elements from @ljbade in #403, and the UDP sample app here: https://github.com/swift-nav/piksi_buildroot/commit/e9872ebc487bd2fb3c22488254f6dea932a36d24.  This expanded sample app also adds a few example settings, and removes the modifications to the `sbp_router.yml` so that we can make a more measured decision around whether to support message "interception".  Adding services to monit no longer require a change to the monitrc, but rather a creating of a `/etc/monitrc.d/service.monitrc` file.

Also addresses https://github.com/swift-nav/firmware_team_planning/issues/386 -- the BR2_BUILD_SAMPLE_DAEMON environment variable must be set to "y" in order for the sample_daemon to be built and installed.

Other changes:
+ `board/piksiv3/post_image.sh` reduce noise and make sure that the path to the built images are dumped at the conclusion of the script

+ Add `awscli` to the docker image so that `make firmware` can be run inside the docker container

+ Change all logging to use sbp_log instead of piksi_log since we're excepting SDK users to use this logging mechanism more heavily (this required improving sbp_log so that it can format messages)

+ Re-enable the monit control port (but use a unix socket instead so that it's not externally accessible, and so that it doesn't trigger monit start-up bugs, e.g. https://bitbucket.org/tildeslash/monit/issues/559/http-service-cant-be-started-on-boot).

+ Add a Python example of how to receive the UDP data and parse it: [udp_receive.py](https://github.com/swift-nav/piksi_buildroot/blob/sdk_sample_app/package/sample_daemon/udp_receive.py)

+ Bug-fix the new docker-wipe target, add some "semi-permanence" of command line history to the docker container

## Testing

+ Bench tested the BR2_BUILD_SAMPLE_DAEMON environment variable to confirm that it adds the `sample_daemon` file to the image the supporting files.

+ Downloaded the firmware image built by Travis and confirmed that the sample_daemon is not built by default.

### CODE REVIEW

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017120901-7-g94644f8
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-7-g94644f8/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017120901-7-g94644f8/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-7-g94644f8

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017120901-11-ge528cf2
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-11-ge528cf2/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017120901-11-ge528cf2/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-11-ge528cf2

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017120901-11-ge528cf2
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-11-ge528cf2/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017120901-11-ge528cf2/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-11-ge528cf2

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017120901-11-ge528cf2
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-11-ge528cf2/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017120901-11-ge528cf2/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-11-ge528cf2

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017120901-11-ge528cf2
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-11-ge528cf2/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017120901-11-ge528cf2/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-11-ge528cf2

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017120901-11-ge528cf2
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-11-ge528cf2/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017120901-11-ge528cf2/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-11-ge528cf2

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017120901-23-g81b5e03
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-23-g81b5e03/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017120901-23-g81b5e03/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-23-g81b5e03

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017120901-21-gf479aa6
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-21-gf479aa6/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017120901-21-gf479aa6/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-21-gf479aa6

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017120901-23-g81b5e03
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-23-g81b5e03/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017120901-23-g81b5e03/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-23-g81b5e03

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017120901-28-gad6f1e3
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-28-gad6f1e3/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017120901-28-gad6f1e3/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017120901-28-gad6f1e3

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017121321-18-gd3eabd9
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017121321-18-gd3eabd9/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017121321-18-gd3eabd9/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017121321-18-gd3eabd9

File: board/piksiv3/post_image.sh [hunk 1]
```diff
@@ -30,46 +40,100 @@ UBOOT_PROD_DIR=$UBOOT_BASE_DIR/build/${CFG}_prod
 UBOOT_FAILSAFE_DIR=$UBOOT_BASE_DIR/build/${CFG}_failsafe
 UBOOT_DEV_DIR=$UBOOT_BASE_DIR/build/${CFG}_dev
 
+LOG_FILE=$(mktemp)
+trap 'rm -rf $LOG_FILE' EXIT
+
 generate_dev() {
+
+  echo -n "Generating DEV firmware image image... "
```
Comment (cbeighley) [inline board/piksiv3/post_image.sh]: image image?

**Comment by cbeighley** [comment]
@silverjam Thanks for getting this up! Here are my comments:
- How does one include the sample app in the build? Can you document this in the top level README.md?
- Do the sample app log messages only come over UDP? I think it would be nice if they came out any interface that had MSG_LOG in the whitelist, so a customer could try the sample app the first time with their normal workflow (probably using the Swift Console over USB-RS232).

File: board/piksiv3/post_image.sh (hunk 1 above)
Comment (silverjam) [reply board/piksiv3/post_image.sh]: Fixed

**Comment by silverjam** [comment]
@cbeighley Will add a section to the top-level README, also planning to re-arrange the sections so that the docker commands are first.  The log messages come over any interface (but only HEARTBEAT and POS_LLS is sent to UDP), so I was using the console's JSON out to debug the daemon, using something like this:

```
tail -f ~/SwiftNav/swift-gnss-20171212-223150.sbp.json | \
  jq --unbuffered -r 'select(.data.msg_type==1025) | .data.text' | \
  grep sample_daemon
```

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017121321-22-g6f79aad
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017121321-22-g6f79aad/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017121321-22-g6f79aad/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017121321-22-g6f79aad

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017121321-22-g6f79aad
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017121321-22-g6f79aad/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017121321-22-g6f79aad/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017121321-22-g6f79aad

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017121321-24-ge4c4287
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017121321-24-ge4c4287/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017121321-24-ge4c4287/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017121321-24-ge4c4287

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017121321-24-ge4c4287
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017121321-24-ge4c4287/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017121321-24-ge4c4287/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017121321-24-ge4c4287

**Comment by cbeighley** [comment]
@silverjam LGTM, thanks for getting this together.

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017121321-25-ga3899fc
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017121321-25-ga3899fc/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017121321-25-ga3899fc/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017121321-25-ga3899fc

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017121321-25-ga3899fc
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017121321-25-ga3899fc/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017121321-25-ga3899fc/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017121321-25-ga3899fc

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017121321-27-gfba5b3d
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017121321-27-gfba5b3d/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017121321-27-gfba5b3d/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017121321-27-gfba5b3d

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2017121321-27-gfba5b3d
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017121321-27-gfba5b3d/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2017121321-27-gfba5b3d/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2017121321-27-gfba5b3d

**Pull request merged**